### PR TITLE
Gulp upgrade to v4

### PR DESCRIPTION
**Pull request by JMRhodes**
- package updates 
- php codesniffer format

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -27,148 +33,112 @@ var PATHS = {
...
-    .pipe($.concat('vendor.js'))
-    .pipe($.rename({
-      suffix: '.min'
-    }))
-    .pipe($.uglify().on('error', swallowError))
-    .pipe(gulp.dest('build/js'));
-});
-
-// Concatenate & Minify all theme javascript files
-gulp.task('dev:scripts:vendor', function () {
```
Comment (sreigle) [inline gulpfile.js]: You will probably want to preserve `dev` scripts in some way. Either as a separate script or an environment variable. Minifying JS in development is much slower to recompile and harder to debug.

File: gulpfile.js [hunk 2]
```diff
@@ -27,148 +33,112 @@ var PATHS = {
...
-    .pipe($.rename({
-      basename: "theme",
-      suffix: '.min'
-    }))
-    .pipe($.sourcemaps.write('.'))
-    .pipe(gulp.dest('build/css'))
-});
-
-// Compile uncompressed CSS
-gulp.task('dev:styles', function () {
```
Comment (sreigle) [inline gulpfile.js]: Same comment for styles, also we only need `sourcemaps` for development not production. Not sure why it is backwards in the old version.

Comment (JMRhodes) [reply gulpfile.js]: We don't currently have a build script when deploying to Pantheon, so the build/dev tasks don't necessarily need to be different for styles at this time.

File: gulpfile.js (hunk 1 above)
Comment (JMRhodes) [reply gulpfile.js]: Good point, I'll add one to make debugging easier.

Comment (JMRhodes) [reply gulpfile.js]: Updated.

Comment (sreigle) [reply gulpfile.js]: 👍

File: gulpfile.js (hunk 2 above)
Comment (sreigle) [reply gulpfile.js]: I think there is value in seeing un-minified css when working locally, but I will leave it up to you.

Comment (JMRhodes) [reply gulpfile.js]: I could see that, however with the sourcemaps we never look at the enqueued CSS file anyways, we're always debugging directly with `scss` files.

Comment (JMRhodes) [reply gulpfile.js]: I'd be fine doing un-minified on development, but that may be more of a phase 2, once we get a different build workflow setup.

**Comment by JMRhodes** [comment]
@arippberger phpcbf task added to gulp.

**Pull request merged**